scvi/module/base/_base_module.py

```diff
@@ -533,8 +556,147 @@ def loss(self, *args, **kwargs) -> LossRecorder:
+
+    def train(self):
+        """Switch to train mode. Emulates Pytorch's interface."""
+        self.training = True
+
     def eval(self):
-        """No-op for PyTorch compatibility."""
+        """Switch to evaluation mode. Emulates Pytorch's interface."""
+        self.training = False
+
+    @property
+    def rngs(self) -> Dict[str, jnp.ndarray]:
+        """
+        Dictionary of RNGs mapping required RNG name to RNG values.
+
+        Calls ``self._split_rngs()`` resulting in newly generated RNGs on
+        every reference to ``self.rngs``.
+        """
+        return self._split_rngs()
+
+    def _set_rngs(self):
+        """Creates RNGs split off of the seed RNG for each RNG required by the module."""
+        required_rngs = self.required_rngs
+        rng_keys = random.split(self.seed_rng, num=len(required_rngs) + 1)
+        self.seed, module_rngs = rng_keys[0], rng_keys[1:]
+        self._rngs = {k: module_rngs[i] for i, k in enumerate(required_rngs)}
+
+    def _split_rngs(self):
+        """
+        Regenerates the current set of RNGs and returns newly split RNGs.
+
+        Importantly, this method does not reuse RNGs in future references to ``self.rngs``.
+        """
+        new_rngs = {}
+        ret_rngs = {}
+        for k, v in self._rngs.items():
+            new_rngs[k], ret_rngs[k] = random.split(v)
+        self._rngs = new_rngs
+        return ret_rngs
+
+    @property
+    def params(self) -> FrozenDict[str, Any]:  # noqa: D102
+        self._check_train_state_is_not_none()
+        return self.train_state.params
+
+    @property
+    def state(self) -> FrozenDict[str, Any]:  # noqa: D102
+        self._check_train_state_is_not_none()
+        return self.train_state.state
+
+    def state_dict(self) -> Dict[str, Any]:
+        """Returns a serialized version of the train state as a dictionary."""
+        self._check_train_state_is_not_none()
+        return flax.serialization.to_state_dict(self.train_state)
+
+    def load_state_dict(self, state_dict: Dict[str, Any]):
+        """Load a state dictionary into a train state."""
+        if self.train_state is None:
+            raise RuntimeError(
+                "Train state is not set. Train for one iteration prior to loading state dict."
+            )
+        self.train_state = flax.serialization.from_state_dict(
+            self.train_state, state_dict
+        )
+
+    def to(self, device: Device):
+        """Move module to device."""
+        self._check_train_state_is_not_none()
+        self.train_state = jax.tree_util.tree_map(
+            lambda x: jax.device_put(x, device), self.train_state
+        )
+        self.seed_rng = jax.device_put(self.seed_rng, device)
+        self._rngs = jax.device_put(self._rngs, device)
+
+    def _check_train_state_is_not_none(self):
+        if self.train_state is None:
+            raise RuntimeError("Train state is not set. Module has not been trained.")
+
+    def as_bound(self) -> JaxBaseModuleClass:
```

Comment:
do we need this?

Reply:
I think it's fine to leave public and keep it, it will be helpful for debugging things